ui/imports/shared/popups/SignTransactionModal.qml

```diff
@@ -129,6 +129,7 @@ StatusModal {
                 RecipientSelector {
                     id: selectRecipient
                     visible: false
+                    ensModule: root.store.contactsModuleInst
```

Comment:
@jrainville you named it `ensModule`, but assigned contacts module to it (that naming thing is not a big deal, but we should taking care about naming, cause that helps a reader a lot in his understanding of our intention). Also I know what was the reason why you used `contactsModule` in this scope, going to explain below.

There is a much bigger problem here, which I am not sure (only because of time) whether to do it now or leave it for later (personally I don't like to left such things, but we may discuss about this), actually the thing is in mixing concepts, so in no way we should access contacts module from here (or from anywhere in the app but the contacts related view which is a subsection of the profile settings section).

What would be a good approach here is to introduce a new module `transactionModule` which will be direct submodule of the main module (since it will be used from the chat/community section, browser section, maybe wallet section) and use it in all transaction related modals. That transaction module will have access to contacts, ens, eth, wallet, accounts and other necessary services for it, but at least one issue that I see at the moment is that from some reason a SSOT (Single Source Of Truth) for `contactList`, `contactRequests`, `addedContacts` and `blockedContacts` is somehow `contactsModule` form the `profileSectionModule`, but not `contact service`, that need to be changed first. 

The point is to keep all contacts related things in the contacts service, and use that service on every place we need it (other services, modules), and contact module should keep only what is needed for contacts subsection of the profile section, nothing else.

@jrainville I am adding this as a comment here (not an issue) and we need to discuss when will be a time to sort this out, but keeping it like this will just bring more confusions later and mixed concepts.

Reply:
@saledjenic I agree with your points and you are right in your assessment of why I pass `ensModule` the `contactsModule`. I know it's pretty ugly and can be confusing, but indeed, our only source of truth for contacts right now is the ContactsModule from the Profile.

I agree that we should move that module to a more shared place, because it is used by a lot of different sections (wallet, chat, profile).

As for the ENS resolving, having a shared ENS or transaction module would help.

For both of those issues, I think it would be simpler to do it in another ticket/PR, because it doesn't really belong with the chat commands PR. I did the ENS fix because using the TX modal just didn't work otherwise, but I could have done it in another one anyway. Also, with me going on vacation for the holidays, I'd never have time to finish that and have it merged, so this PR would just get behind and conflict.

If you give me your thumbs up, I'll add the two tickets and merge this.